### ISSUE TITLE
fix(spherical): `SphericalVorticity2D.potential_vorticity` with a mask has NaN gradients (missing the denominator guard `Vorticity2D` has)

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/finitevolX#287]
Part of #275.

## Summary

`SphericalVorticity2D.potential_vorticity` (`finitevolx/_src/operators/spherical_compound.py:155-181`) says in its docstring "same fix as `Vorticity2D.potential_vorticity`", but the guard `den = jnp.where(mask.xy_corner_strict[1:-1, 1:-1], den, 1.0)` (`vorticity.py:136`) is missing. The forward pass is finite (`jnp.where(den == 0, nan, num/den)` selects the safe branch) but reverse mode produces `0 · inf` at every dry corner.

## Reproduction

```python
import jax; jax.config.update("jax_enable_x64", True)
import jax.numpy as jnp, numpy as np
from finitevolx import SphericalGrid2D, SphericalVorticity2D, Mask2D
g = SphericalGrid2D.from_interior(30, 14, (0., 360.), (-60., 60.))
m = np.ones((g.Ny, g.Nx), bool); m[[0, -1], :] = False; m[:, [0, -1]] = False; m[5:9, 10:14] = False
mask = Mask2D.from_mask(jnp.asarray(m))
op = SphericalVorticity2D(g, mask=mask)
u = jnp.ones((g.Ny, g.Nx)); v = jnp.zeros_like(u); f = jnp.ones_like(u); h = 10.0 + jnp.zeros_like(u)
pv = op.potential_vorticity(u, v, h, f)
gh = jax.grad(lambda hh: op.potential_vorticity(u, v, hh, f).sum())(h)
print("forward NaN count:", int(jnp.isnan(pv).sum()), " grad NaN count:", int(jnp.isnan(gh).sum()))
```

```
SphericalVorticity2D(mask).pv: NaN count in d(sum pv)/dh = 56 ; forward NaN count = 0
Vorticity2D(mask).pv        : NaN count in d(sum pv)/dh = 0
```

## Proposed fix

Copy the two-line guard from `Vorticity2D.potential_vorticity`: replace the denominator by 1.0 on dry corners before dividing, then mask the output. Consider factoring the guarded division into one helper used by both classes.

## Acceptance criteria

- [ ] `jax.grad` of the masked spherical PV w.r.t. `h`, `u`, `v` is finite (test).
- [ ] Forward values unchanged on wet corners.
